gravitee-am-ui/src/app/domain/applications/application/advanced/factors/step-up-auth/mfa-step-up.component.ts

```diff
@@ -45,4 +45,23 @@ export class MfaStepUpComponent {
       },
     });
   }
+
+  onToggle($event: any): void {
+    this.stepUpAuth.active = $event.checked;
+    this.update();
+  }
+
+  private update(): void {
+    if (this.stepUpAuth.active) {
+      this.settingsChange.emit({
```

Comment:
I think it can be just 

 this.settingsChange.emit(stepUpAuth);

Reply:
right...